pkg/ext/store.go

```diff
@@ -29,6 +31,25 @@ type Context struct {
 	GroupVersionResource schema.GroupVersionResource
 }
 
+type CreateFunc[T runtime.Object] func(ctx Context, obj T, opts *metav1.CreateOptions) (T, error)
+type UpdateFunc[T runtime.Object] func(ctx Context, obj T, opts *metav1.UpdateOptions) (T, error)
+type GetFunc[T runtime.Object] func(ctx Context, name string, opts *metav1.GetOptions) (T, error)
+type ListFunc[TList runtime.Object] func(ctx Context, opts *metav1.ListOptions) (TList, error)
+type WatchFunc[T runtime.Object] func(ctx Context, opts *metav1.ListOptions) (<-chan WatchEvent[T], error)
+type DeleteFunc[T runtime.Object] func(ctx Context, name string, opts *metav1.DeleteOptions) error
+
+// ConvertFunc will convert a object to a list of cell in a metav1.Table (think kubectl get table output)
+type ConvertFunc[T runtime.Object] func(obj T) []string
+
+type Storage[T runtime.Object, TList runtime.Object] interface {
```

Comment:
Now we have a Store interface and a Storage interface. The old store interface forced us to implement all operations even if not necessary, so we'd like to move away from that.

I found the old store interface pretty convenient for tests because I'm building many different stores. I made it work with the new API with the new "StandardStore" struct, which takes in a Store. Maybe instead I should just remove the old Store interface just do:

`NewStandardStore(listFn, getFn, etc)` which would make it the same as the old store..